### PR TITLE
[Metrics][LOG] Update metrics of 'max_compaction_score' and log for compaction

### PR DESCRIPTION
**Pull request by weizuo93**
## Proposed changes

1. Update metrics of `tablet_base_max_compaction_score` and `tablet_cumulative_max_compaction_score`. This two metrics should show `max compaction score` for base comapction and cumulative compaction after a round of task production in BE.
2. Update log to show current max version after each compaction task finished so that we can see whether compaction can keep up with data load.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

File: be/src/olap/olap_server.cpp [hunk 1]
```diff
@@ -386,20 +386,30 @@ void StorageEngine::_compaction_tasks_producer_callback() {
...
             continue;
         }
         if (!data_dir->reach_capacity_limit(0)) {
+            uint32_t disk_max_score = 0;
             TabletSharedPtr tablet = _tablet_manager->find_best_tablet_to_compaction(
-                    compaction_type, data_dir, _tablet_submitted_compaction[data_dir]);
+                    compaction_type, data_dir, _tablet_submitted_compaction[data_dir], disk_max_score);
             if (tablet != nullptr) {
                 tablets_compaction.emplace_back(tablet);
+                max_compaction_score = disk_max_score > max_compaction_score ? disk_max_score : max_compaction_score;
```
Comment (acelyc111) [inline be/src/olap/olap_server.cpp:402]:
```suggestion
                max_compaction_score = std::max(max_compaction_score, disk_max_score);
```

File: be/src/olap/olap_server.cpp [hunk 2]
```diff
@@ -386,20 +386,30 @@ void StorageEngine::_compaction_tasks_producer_callback() {
...
             }
         }
     }
+    if (tablets_compaction.size() > 0) {
```
Comment (acelyc111) [inline be/src/olap/olap_server.cpp:406]:
```suggestion
    if (!tablets_compaction.empty()) {
```
or `max_compaction_score > 0` ？

File: be/src/olap/tablet_manager.h [hunk 3]
```diff
@@ -71,7 +71,7 @@ class TabletManager {
 
     TabletSharedPtr find_best_tablet_to_compaction(CompactionType compaction_type,
                                                    DataDir* data_dir,
-                                                   vector<TTabletId>& tablet_submitted_compaction);
+                                                   vector<TTabletId>& tablet_submitted_compaction, uint32_t& score);
```
[the remark this answers is not in the document]
Comment (weizuo93) [reply be/src/olap/tablet_manager.h]: OK.

File: be/src/olap/olap_server.cpp (hunk 1 above)
Comment (weizuo93) [reply be/src/olap/olap_server.cpp]: OK.

File: be/src/olap/olap_server.cpp (hunk 2 above)
Comment (weizuo93) [reply be/src/olap/olap_server.cpp]: OK.

File: be/src/olap/tablet_manager.cpp [hunk 4]
```diff
@@ -681,7 +681,7 @@ void TabletManager::get_tablet_stat(TTabletStatResult* result) {
 
 TabletSharedPtr TabletManager::find_best_tablet_to_compaction(
         CompactionType compaction_type, DataDir* data_dir,
-        std::vector<TTabletId>& tablet_submitted_compaction) {
+        std::vector<TTabletId>& tablet_submitted_compaction, uint32_t& score) {
```
Comment (yangzhg) [inline be/src/olap/tablet_manager.cpp:684]: use  a pointer instead

Comment (weizuo93) [reply be/src/olap/tablet_manager.cpp:684]: > use a pointer instead

OK, Thank you.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**